2017-02-08  Jason Andrew Less  <less@lnxsrv09.seas.ucla.edu>

	* timespec.h(timespec_cmp): Due to the possibility that overflow could occur
	when the difference between the values of a and b was too large, I used a
	direct comparison alternative, instead of taking the difference of the two.
	

--- timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ /u/cs/ugrad/less/timespec.h	2017-02-08 23:05:47.081016000 -0800
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec < b.tv_sec)
+    return -1;
+  else if (a.tv_sec > b.tv_sec)
+    return 1;
+  else // (a.tv_sec == b.tv_sec)
+    return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
